### PR TITLE
ztest: cleanup phase should reset the timeout queue.

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
Reusing the k_thread structure requires a cleanup of all essential
info. We need to remove the ztest_thread information from the timeout_q.

If left untouched, when a tick occurs the thread's delta_ticks_from_prev
would be corrupted. This inturn causes a chain reaction of problems.
Thus once the unit test is completed the timeout_q is scrubbed.

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

### CODE REVIEW

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
I don't think this is the fix we want.
We are basically playing around directly with internal kernel data structures, from the application, so that thread objects may be reused.
Under what circumstances does this happen anyway?
Why don't we make adjustments to the timeout queue in k_thread_abort instead?

**Comment by AdithyaBaglody** [comment]
A unit test creates a new thread with the k_tid_t  ztest_thread. From there are I create another new thread(say foo). Which does some tests and tells ztest_pass(). Now the ztest_pass() will kill the new thread( i.e foo) and it doesn't kill the ztest_thread. So by the time this ztest_thread comes back to the entry function and call the abort on itself we would have reinitialized the k_thread stuct. so an abort never gets called.

We cant handle this in any kernel function because it a problem of the z test implementation.
So I understand that I shouldn't touch the data structures from ztest. I will call k thread abort on the k_tid_t ztest_thread during cleanup.

File: tests/ztest/src/ztest.c [hunk 1]
```diff
@@ -23,6 +26,15 @@ static int cleanup_test(struct unit_test *test)
 	int mock_status;
 
 	mock_status = _cleanup_mock();
+
+	#ifdef KERNEL
+	/* we need to remove the ztest_thread information from the timeout_q.
+	 * Because we reuse the same k_thread structure this would
+	 * causes some problems.
+	 */
```
Comment (SavinayDharmappa) [inline tests/ztest/src/ztest.c]: may be you have change your comment. I don't see a you are directly touching timeout queue and even your commit message as well.

File: tests/ztest/src/ztest.c [hunk 2]
```diff
@@ -23,6 +26,15 @@ static int cleanup_test(struct unit_test *test)
 	int mock_status;
 
 	mock_status = _cleanup_mock();
+
+	#ifdef KERNEL
```
Comment (andrewboie) [inline tests/ztest/src/ztest.c]: can you get rid of the leading whitespace before "#"?

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
style nit.
substance of the patch looks good.

**Comment by AdithyaBaglody** [comment]
Delta:
Handled Andrew's review comments.

**Comment by punitvara** [comment]
LGTM

**Review by nashif — APPROVED** [review]

**Pull request merged**